File: src/main/java/com/jxdinfo/hussar/hussardemo/controller/ChangeToPinyinDemo.java
/*
 * 金现代轻骑兵V8开发平台 
 * ChangeToPinyinDemo.java 
 * 版权所有：金现代信息产业股份有限公司  Copyright (c) 2018-2023 .
 * 金现代信息产业股份有限公司保留所有权利,未经允许不得以任何形式使用.
 */
package com.jxdinfo.hussar.hussardemo.controller;

import com.jxdinfo.hussar.common.projecttool.PinYinUtil;
import com.jxdinfo.hussar.core.base.controller.BaseController;
import net.sourceforge.pinyin4j.format.HanyuPinyinCaseType;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;

/**
 * 类的用途：汉字转拼音Demo
 * 创建日期：2018年10月17日
 * 修改历史：
 * 修改日期：2018年10月17日
 * 修改作者：XinXueLei
 * 修改内容：修改内容
 * @author XinXueLei
 * @version 1.0
 */
@RequestMapping("/changeToPinyinDemo")
@Controller
public class ChangeToPinyinDemo extends BaseController {
    /**
     *
     * 汉字转拼音demo页面
     * @Title: view
     * @author: XinXueLei
     * @return 页面
     */
    @RequestMapping("/view")
    public String view(Model model) {
        return "/hussardemo/changeToPinyinDemo.html";
    }

    /**
     *
     * 将汉字转为拼音，全部大写
     * @author: XinXueLei
     * @return String
     */
    @RequestMapping("/changeToPinyinUp")
    @ResponseBody
    public String changeToPinyinUp() {
        String beforeChangeStr = super.getPara("beforeChangeOne");
        String afterChangeStr = PinYinUtil.changeToPinyin(beforeChangeStr, HanyuPinyinCaseType.UPPERCASE);
        return afterChangeStr;
    }

    /**
     *
     * 将汉字转为拼音，全部小写
     * @author: XinXueLei
     * @return String
     */
    @RequestMapping("/changeToPinyinLow")
    @ResponseBody
    public String changeToPinyinLow() {
        String beforeChangeStr = super.getPara("beforeChangeTwo");
        String afterChangeStr = PinYinUtil.changeToPinyin(beforeChangeStr, HanyuPinyinCaseType.LOWERCASE);
        return afterChangeStr;
    }

    /**
     *
     * 获取每个汉字的首字母大写
     * @author: XinXueLei
     * @return String
     */
    @RequestMapping("/getFirstLettersUp")
    @ResponseBody
    public String getFirstLettersUp() {
        String beforeChangeStr = super.getPara("beforeChangeThree");
        String afterChangeStr = PinYinUtil.getFirstLetters(beforeChangeStr,HanyuPinyinCaseType.UPPERCASE);
        return afterChangeStr;
    }

    /**
     *
     * 获取每个汉字的首字母小写
     * @author: XinXueLei
     * @return String
     */
    @RequestMapping("/getFirstLettersLow")
    @ResponseBody
    public String getFirstLettersLow() {
        String beforeChangeStr = super.getPara("beforeChangeFour");
        String afterChangeStr = PinYinUtil.getFirstLetters(beforeChangeStr,HanyuPinyinCaseType.LOWERCASE);
        return afterChangeStr;
    }
}

File: src/main/webapp/static/js/hussardemo/changeToPinyinDemo.js
/**
 * @Description: 汉字转拼音Demo脚本文件
 * @Author: xinxuelei
 * @Date: 2018/10/17
 */
layui.use(['jquery','layer','Hussar','HussarAjax','form','element'], function(){
    var Hussar = layui.Hussar;
    var $ = layui.jquery;
    var layer = layui.layer;
    var $ax = layui.HussarAjax;
    var form = layui.form;
    var element = layui.element;

    /**
     * 汉字转拼音Demo的单例
     */
    var changeToPinyinDemo = {
        layerIndex: -1
    };

    /**
     *  初始化按钮事件
     */
    changeToPinyinDemo.initButtonEvnts = function () {
        $("#changeToPinyinUp").click(function(){
            changeToPinyinDemo.changeToPinyinUp();
        });
        $("#changeToPinyinLow").click(function(){
            changeToPinyinDemo.changeToPinyinLow();
        });
        $("#getFirstLettersUp").click(function(){
            changeToPinyinDemo.getFirstLettersUp();
        });
        $("#getFirstLettersLow").click(function(){
            changeToPinyinDemo.getFirstLettersLow();
        });
    }

    /**
     *  将汉字转为拼音，全部大写
     */
    changeToPinyinDemo.changeToPinyinUp = function () {
        var beforeChangeStr = $('#beforeChangeOne').val();
        if("" != beforeChangeStr && null != beforeChangeStr){
            var ajax = new $ax(Hussar.ctxPath + "/changeToPinyinDemo/changeToPinyinUp",
                function(data) {
                    if(data.length==0){
                        $("#afterChangeOne").val("无");
                    }else{
                        $("#afterChangeOne").val(data);
                    }
                },
                function(data) {
                    Hussar.error("汉字转拼音转换失败!");
                });
            ajax.set("beforeChangeOne",beforeChangeStr);
            ajax.start();
        }else{
            Hussar.info("待转换内容不能为空！");
        }
    }

    /**
     *  将汉字转为拼音，全部小写
     */
    changeToPinyinDemo.changeToPinyinLow = function () {
        var beforeChangeStr = $('#beforeChangeTwo').val();
        if("" != beforeChangeStr && null != beforeChangeStr) {
            var ajax = new $ax(Hussar.ctxPath + "/changeToPinyinDemo/changeToPinyinLow",
                function (data) {
                    if (data.length == 0) {
                        $("#afterChangeTwo").val("无");
                    } else {
                        $("#afterChangeTwo").val(data);
                    }
                },
                function (data) {
                    Hussar.error("汉字转拼音转换失败!");
                });
            ajax.set("beforeChangeTwo", beforeChangeStr);
            ajax.start();
        }else{
            Hussar.info("待转换内容不能为空！");
        }
    }

    /**
     *  获取每个汉字的首字母大写
     */
    changeToPinyinDemo.getFirstLettersUp = function () {
        var beforeChangeStr = $('#beforeChangeThree').val();
        if("" != beforeChangeStr && null != beforeChangeStr) {
            var ajax = new $ax(Hussar.ctxPath + "/changeToPinyinDemo/getFirstLettersUp",
                function (data) {
                    if (data.length == 0) {
                        $("#afterChangeThree").val("无");
                    } else {
                        $("#afterChangeThree").val(data);
                    }
                },
                function (data) {
                    Hussar.error("汉字转拼音转换失败!");
                });
            ajax.set("beforeChangeThree", beforeChangeStr);
            ajax.start();
        }else{
            Hussar.info("待转换内容不能为空！");
        }
    }

    /**
     *  获取每个汉字的首字母小写
     */
    changeToPinyinDemo.getFirstLettersLow = function () {
        var beforeChangeStr = $('#beforeChangeFour').val();
        if("" != beforeChangeStr && null != beforeChangeStr) {
            var ajax = new $ax(Hussar.ctxPath + "/changeToPinyinDemo/getFirstLettersLow",
                function (data) {
                    if (data.length == 0) {
                        $("#afterChangeFour").val("无");
                    } else {
                        $("#afterChangeFour").val(data);
                    }
                },
                function (data) {
                    Hussar.error("汉字转拼音转换失败!");
                });
            ajax.set("beforeChangeFour", beforeChangeStr);
            ajax.start();
        }else{
            Hussar.info("待转换内容不能为空！");
        }
    }

    $(function () {
        changeToPinyinDemo.initButtonEvnts();  //初始化按钮事件
    });
});